### PR TITLE
feat: modifed kafka to use config received from config-be.

### PR DESCRIPTION
**Pull request by saurav-malani**
# Description
Modified kafka to get ssh config from config-BE in addition to from env. 
NOTE: getting config via env has to be deprecated in future. 

## Notion Ticket

https://www.notion.so/rudderstacks/Kafka-with-SSH-control-plane-config-8a25b92171464903a2a5db9e40645290?pvs=4

## Security

- [ ] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

### CODE REVIEW

File: services/streammanager/kafka/kafkamanager.go
Comment (fracasula) [inline services/streammanager/kafka/kafkamanager.go:329]: Can you please create a task on Notion to follow up on this?

Comment (saurav-malani) [reply services/streammanager/kafka/kafkamanager.go:329]: https://www.notion.so/rudderstacks/kafka-over-ssh-via-env-cleanup-after-control-plane-changes-are-merged-b1d1d074073d4d398aab083f6d153ee8?pvs=4

File: services/streammanager/kafka/kafkamanager_test.go
Comment (saurav-malani) [reply services/streammanager/kafka/kafkamanager_test.go:120]: test for missing complete sshConfig is already added. Anyways I have added one more test where except `sshUser` everything is there. And regarding happy case, is is indirectly added in the existing happy case `ok`.

Apologies if I misunderstood something.

File: services/controlplane/client.go
Comment (fracasula) [inline services/controlplane/client.go:70]: @saurav-malani renamed this to `SSHKeyPair` as it's more evocative.

File: services/controlplane/client.go
Comment (fracasula) [reply services/controlplane/client.go:116]: Additional changes here were made to have better error messages in this method.

File: services/controlplane/client_test.go
Comment (fracasula) [inline services/controlplane/client_test.go:398]: @saurav-malani the public key was not being tested, I added it now.

File: services/controlplane/client_test.go
Comment (fracasula) [inline services/controlplane/client_test.go:350]: @saurav-malani I added this to test the retries behaviour. The first call always fails.

File: services/controlplane/client_test.go
Comment (fracasula) [inline services/controlplane/client_test.go:347]: @saurav-malani I converted some test checks to make use of `require` for a more succint test.

File: services/streammanager/kafka/kafkamanager.go
Comment (fracasula) [inline services/streammanager/kafka/kafkamanager.go]: @saurav-malani I moved the SSH block out of the `if destConfig.SslEnabled {` condition like it was before.

File: services/streammanager/kafka/client/client.go
Comment (fracasula) [inline services/streammanager/kafka/client/client.go:79]: @saurav-malani as discussed over standup this is an important change I introduced (cc @lvrach) and it's going to apply to all SSH connections for kafka destinations.

**Review by fracasula — APPROVED** [review]

**Comment by fracasula** [comment]
Let's not merge this one until we have the chance to do some proper manual end to end testing on dev.

**Review by lvrach — APPROVED** [review]

**Review by fracasula — APPROVED** [review]
@saurav-malani I forced push a local copy that I had of the branch before you forced pushed the commit I co-authored. Please leave these changes as they are, we just need to test them. If you need to test other things I suggest you start from this branch and push a copy of it e.g. `feat.kafkaOverSshViaCP-test`.

**Comment by fracasula** [comment]
@saurav-malani I'll resolve the conflicts, don't worry :+1:

**Pull request merged**
